wireapp/wire-android
# fix: infinite call vibration (AR-3214)

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-3214" title="AR-3214" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />AR-3214</a>  PlayTest 14.03 - Phone keeps vibrating when rejecting an incoming call
  </td></table>
  <br />
 










----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Notification for incoming call would keep an infinite loop of vibration when caller start the call and cancel before the other user even gets the first calling signal.
The only way to stop it was opening the notification tray.

### Solutions

Add a delay of 300ms before cancelling the notification so it has time to properly set the notification vibration+sound before cancelling it (as it was being called around less than 100ms) and thus not doing exactly what it needed to do (cancel the notification vibration+sound).

Because from out POV it was like this:
`Create Notification -> Create Vibration object -> cancel notification -> start vibration -> infinite vibration`
with the delay we are giving a bit of time for the vibration in notification to properly start and then properly be cancelled.

### Testing

#### How to Test

- Have app in vibration mode
- Have 3 accounts
- Login with acc1 and acc2
- make acc2 active
- call acc1 from acc3 and cancel before acc3 even receives the first call signal
- app should not be stuck in vibration
